### PR TITLE
Back port : Add search content in help dialog and fix document repair for non-Eng UI to 23.05

### PR DESCRIPTION
**Pull request by Darshan-upadhyay1110**
Backport PR of https://github.com/CollaboraOnline/online/pull/8211

* Resolves: # 
* Target version: master 

### TODO

- [ ] ...

### Checklist

- [ ] Code is properly formatted
- [ ] All commits have Change-Id
- [ ] I have run tests with `make check`
- [ ] I have issued `make run` and manually verified that everything looks okay
- [ ] Documentation (manuals or wiki) has been updated or is not required

* Resolves: # 
* Target version: master 

### TODO

- [ ] ...

### Checklist

- [ ] Code is properly formatted
- [ ] All commits have Change-Id
- [ ] I have run tests with `make check`
- [ ] I have issued `make run` and manually verified that everything looks okay
- [ ] Documentation (manuals or wiki) has been updated or is not required

### CODE REVIEW

File: browser/html/cool-help.html [hunk 1]
```diff
@@ -6,573 +6,777 @@
...
+      box-shadow: 0 0 10px rgba(0, 0, 0, 0.2);
+    }
+    #online-help-search-input::placeholder {
+      color: #999;
+    }
     [data-theme='dark'] span.cool-annotation-menu { filter: invert(1);}
 </style>
+
+<div class="search-container">
+  <input size="20" id="online-help-search-input" placeholder="Search..." />
```
Comment (timar) [inline browser/html/cool-help.html:68]: placeholder="Search..." is not localisable.

Comment (Darshan-upadhyay1110) [reply browser/html/cool-help.html:68]: @timar i did not understand your comment.

I guess you are trying to suggest that we should not use `...` in placeholder for any translation errors right ?

Comment (timar) [reply browser/html/cool-help.html:68]: See https://github.com/CollaboraOnline/online/pull/8360
We'll need that commit in co-23.05, too.

**Review by pedropintosilva — APPROVED** [review]

File: browser/src/control/Toolbar.js [hunk 2]
```diff
@@ -451,7 +451,6 @@ L.Map.include({
 			else if (map.getDocType() === 'drawing') {
 				document.getElementById('drawing-shortcuts').style.display='block';
 			}
-			document.getElementById('online-help-search-input').style.display='none';
 		} else /* id === 'online-help' */ {
 			document.getElementById('keyboard-shortcuts-content').style.display='none';
 			document.getElementById('online-help-search-input').setAttribute('placeholder',_('Search'));
```
Comment (pedropintosilva) [inline browser/src/control/Toolbar.js:456]: if we enable it in the keyboard dialog we should also fix the `Search...` non possible to translate string

**Review by pedropintosilva — CHANGES_REQUESTED** [review]
if we enable it in the keyboard dialog we should also fix the Search... non possible to translate string

File: browser/src/control/Toolbar.js (hunk 2 above)
Comment (Darshan-upadhyay1110) [reply browser/src/control/Toolbar.js]: Added new patch which will localize the string for both dialog

**Review by pedropintosilva — APPROVED** [review]
Thanks now I see "Search" in both dialogs and when running make pot inside of browser I correctly see:

```javascript
#: src/control/Control.Menubar.js:928 src/control/Control.Menubar.js:993
#: src/control/Control.Menubar.js:1057 src/control/Control.Menubar.js:1111
#: src/control/Control.NotebookbarCalc.js:887
#: src/control/Control.NotebookbarDraw.js:949
#: src/control/Control.NotebookbarImpress.js:1076
#: src/control/Control.NotebookbarWriter.js:907
#: src/control/Control.StatusBar.js:203 src/control/Control.Zotero.js:221
#: src/control/Toolbar.js:552
msgid "Search"
msgstr ""
```
in the ui pot file

**Comment by Darshan-upadhyay1110** [comment]
CI failure is not related to this PR changes. I have checked cypress test locally 
image

 Rebased PR, waiting for PR to pass...

**Pull request merged**